Improvements fee rates logic

gemdev111 opened the pull request:
Integration GemFeeCalculator (core)
Updates EthereumService & EthereumFeeService
Force updated EthereumFeeHistory
Added EthereumFeeCalculator for GemFeeCalculator usage

ref: https://github.com/gemwalletcom/core/issues/546
https://github.com/gemwalletcom/core/commit/6595a64847c5ebf170978eb5adbe36cd1a453567#diff-b929ed4f6fd34aa4305677a787da7223e4b669b39e607cf7ca630daa488b843a


<img width="1288" alt="image" src="https://github.com/user-attachments/assets/3a2b551c-88ee-4052-b439-3a94ea87d165" />

gemdev111 committed:
Improvements fee rates logic

Integration GemFeeCalculator (core)
Updates EthereumService & EthereumFeeService
Force update EthereumFeeHistory ( needs typeshare update )
Added EthereumFeeCalculator wrapper on GemFeeCalculator

gemdev111 committed:
minor update

gemdev111 committed:
Added protocol constraint

gemdev111 committed:
Minor cleanup

0xh3rman commented on Packages/Blockchain/Sources/Ethereum/EthereumFeeService.swift:
```diff
@@ -251,6 +200,20 @@ extension EthereumService: ChainFeeRateFetchable {
+
+    private func getBasePriorityFees(config: EvmChainConfig) async throws -> (base: BigInt, priority: [FeePriority: BigInt]) {
+        let feeHistory = try await provider
+            .request(.feeHistory(
```
maybe also worth trying to pass `pending` not `latest` when fetch from rpc to see if gives better results

gemdev111 committed:
Merge branch 'main'

gemdev111 committed:
Updated core
Updated EthereumFeeCalculator to support latest core update
Updated get fee history fron node to support differnt block paratameters ( switched from latest to pending )
Cleanup EthereumFeeCalculator

gemdev111 committed:
fix ci, removed unused comma

gemdev111 committed:
removed comma

0xh3rman reviewed (approved):
lgtm

gemcoder21 committed:
Update core

gemcoder21 committed:
Merge branch 'main' into 546-eth-fee-rates-improvements

gemdev111 committed:
Merge branch 'main'
